libsolidity/lsp/LanguageServer.cpp

```diff
@@ -134,7 +134,7 @@ string LanguageServer::clientPathToSourceUnitName(string const& _path) const
 	string path = _path;
 	if (path.find("file://") == 0)
 		path = path.substr(7);
-	// TODO this still uses the CWD in some circumstances.
```

Comment:
It is true, it depends on the base path. At least cwd access code is part of the function.